src/array.jl

```diff
@@ -1,7 +1,7 @@
 """
-    OneHotArray{T,L,N,M,I} <: AbstractArray{Bool,M}
+    OneHotArray{T, L, N, M, I} <: AbstractArray{Bool, M}
 
-These are constructed by [`onehot`](@ref) and [`onehotbatch`](@ref).
+Constructed by [`onehot`](@ref) and [`onehotbatch`](@ref).
 Parameter `I` is the type of the underlying storage, and `T` its eltype.
```

Comment:
```suggestion
A one-hot `M`-dimensional array (with `L` labels)
stored as a compact `N`-dimensional array of indices.
Typically constructed by [`onehot`](@ref) and [`onehotbatch`](@ref).
Parameter `I` is the type of the underlying storage, and `T` its eltype.
```

Reply:
Maybe this should say that `N == M+1` always? And explain that L labels means `size(A,1) == L`?

Reply:
Something like this?

```suggestion
A one-hot `M`-dimensional array (with `L` labels in the first dimension)
stored as a compact `N`-dimensional array of indices (where `N == M + 1`).
Typically constructed by [`onehot`](@ref) and [`onehotbatch`](@ref).
Parameter `I` is the type of the underlying storage, and `T` its eltype.
```

Reply:
But if you don't know what labels means, I think this is still confusing. Maybe:
```suggestion
A one-hot `M`-dimensional array with `L` labels (i.e. `size(A, 1) == L` and `sum(A, dims=1) == 1`)
stored as a compact `N == M-1`-dimensional array of indices.

Typically constructed by [`onehot`](@ref) and [`onehotbatch`](@ref).
Parameter `I` is the type of the underlying storage, and `T` its eltype.
```
Small details, for sure.